qualtran/cirq_interop/_cirq_to_bloq.py

```diff
@@ -53,34 +54,36 @@ def _get_cirq_quregs(signature: Signature, qm: InteropQubitManager):
     return ret
 
 
-@frozen
-class CirqGateAsBloq(GateWithRegisters):
+class CirqGateAsBloqBase(GateWithRegisters):
     """A Bloq wrapper around a `cirq.Gate`"""
 
-    gate: cirq.Gate
+    @property
+    @abc.abstractmethod
+    def cirq_gate(self) -> cirq.Gate:
+        ...
 
     def pretty_name(self) -> str:
-        g = min(self.gate.__class__.__name__, str(self.gate), key=len)
+        g = min(self.cirq_gate.__class__.__name__, str(self.cirq_gate), key=len)
         return f'cirq.{g}'
 
     @cached_property
     def signature(self) -> 'Signature':
-        if isinstance(self.gate, Bloq):
-            return self.gate.signature
-        nqubits = cirq.num_qubits(self.gate)
+        if isinstance(self.cirq_gate, Bloq):
+            return self.cirq_gate.signature
+        nqubits = cirq.num_qubits(self.cirq_gate)
         return (
-            Signature([Register('qubits', shape=nqubits, bitsize=1)])
-            if nqubits > 0
```

Comment:
Is this conditional (> 0) no longer necessary?

Reply:
Correct, because `Signature.build(x=0)` now returns `Signature([])`